api/client/charms/client_test.go

```diff
@@ -300,42 +301,33 @@ func (s *charmsMockSuite) TestListCharmResources(c *gc.C) {
 
 func (s *charmsMockSuite) TestZipHasHooksOnly(c *gc.C) {
 	ch := testcharms.Repo.CharmDir("storage-filesystem-subordinate") // has hooks only
-	tempFile, err := os.CreateTemp(c.MkDir(), "charm")
-	c.Assert(err, jc.ErrorIsNil)
-	defer tempFile.Close()
-	defer os.Remove(tempFile.Name())
-	err = ch.ArchiveTo(tempFile)
+	charmPath := filepath.Join(c.MkDir(), "charm")
+	err := ch.ArchiveToPath(charmPath)
 	c.Assert(err, jc.ErrorIsNil)
 	f := *charms.HasHooksOrDispatch
```

Comment:
**thought:** That's... disturbing. I see no reason to write this like that.

Reply:
it is quite odd. However, this PR is pretty broad in changes, so I'm hesitant to sneak in any more quick fixes